### ISSUE TITLE
Pod name with a dot not extracted correctly from tag

### ISSUE THREAD

**Issue by swalkowski** [GoogleCloudPlatform/fluent-plugin-google-cloud#35]
I noticed that @tag_to_kubernetes_labels_regexp does accept dots inside a pod name:

```
@tag_to_kubernetes_labels_regexp =
  /\.(?<pod_name>[^\._]+)_(?<namespace_name>[^_]+)_(?<container_name>.+)$/
```

However, Kubernetes allows me to have a pod with dots inside it. According to a message from Kubernetes, the pod name:

```
must be a DNS subdomain (at most 253 characters, matching regex [a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*): e.g. "example.com"
```

@a-robinson - does this regexp intentionally not accept a dot inside a pod name, or is this a bug?

**Comment by a-robinson** [comment]
Thanks for catching this, @swalkowski. It is certainly an oversight.

The plugin becomes slightly less flexible if we allow dots there, because it'll basically grab everything from the tag except for the first chunk, regardless of how many chunks are actually part of the pod. We may want to switch this to being a config parameter. I'll send a PR

**Comment by a-robinson** [comment]
Also, cc @jimmidyson, since this also affects his metadata collector's regex.

**Comment by jimmidyson** [comment]
Thanks! Fixing in fabric8io/fluent-plugin-kubernetes_metadata_filter

**Comment by jimmidyson** [comment]
Damn this is harder than it should be.

**Comment by jimmidyson** [comment]
So fixed in https://github.com/fabric8io/fluent-plugin-kubernetes_metadata_filter/commit/b7ca287752585f03fc4188d2f69897e1c95814d8

Tag regex is already a config param & uses sane default. Guess you can do the same @a-robinson?

**Comment by a-robinson** [comment]
Will be fixed by #36

**Comment by mr-salty** [comment]
fixed by #36 and released as gem version 0.4.10 and google-fluentd 1.5.4
